Enable pickling via protobuf

naddeoa opened the pull request:
## Description

We were never able to pickle our dataset profile objects. Our
integration work with Ray made this a pressing issue since Ray depends
heavily on calling pickle on objects. This change will enable pickle by
serializing/deserializing to/from protobuf.

In doing this, I stumbled upon other actual issues, like we apparently
don't deserialize from protobuf correctly in certain cases, like when we
have no set dataset_timestamp. It ends up as a 1970 date. I have two
tests that are omitted that I'm going to work on next to fix that
issue.





### General Checklist

* [x] Tests added for this feature/bug
      if it was a bug, test must cover it.
* [x] Conform by the style guides, by using formatter
* [x] (optional) Please add a label to your PR

Commit:
Enable pickling via protobuf

We were never able to pickle our dataset profile objects. Our
integration work with Ray made this a pressing issue since Ray depends
heavily on calling pickle on objects. This change will enable pickle by
serializing/deserializing to/from protobuf.

In doing this, I stumbled upon other actual issues, like we apparently
don't deserialize from protobuf correctly in certain cases, like when we
have no set dataset_timestamp. It ends up as a 1970 date. I have two
tests that are omitted that I'm going to work on next to fix that
issue.

jamie256 commented on src/whylogs/core/datasetprofile.py:
```diff
@@ -110,6 +110,15 @@ def __init__(
+
+    def __setstate__(self, serizlied_profile):
+        profile = DatasetProfile.parse_delimited_single(serizlied_profile)[1]
+        print("inside")
```
remove print statements, you can put some diagnostic message in a logger as debug level if you think that would help.

andyndang replied:
print debugging

jamie256 reviewed (commented):
some minor comments

jamie256 commented:
Nice find on the time deserialization issues. I'll review again when those fixes are in and update here.

Commit:
Fix some date serialization bugs

Found a few issues around protobuf serde with dates that would cause the
deserialized profiles to have 1970 dataset timestamps or have otherwise
subtlyl different datetime state.

jamie256 reviewed (approved):
Looks good!
